src/vivliostyle/repetitiveelements.js

```diff
@@ -0,0 +1,925 @@
+            if (nodeContext.sourceNode === n) {
+                return nodeContext.after;
+            } else {
+                parentsOfLastContent.push(n);
+            }
+        }
+        for (var currentParent=nodeContext.sourceNode; currentParent; currentParent=currentParent.parentNode) {
+            if (parentsOfLastContent.indexOf(currentParent) >= 0) {
+                return false;
+            } else {
+                for (var current=currentParent; current; current=current.previousElementSibling) {
+                    if (parentsOfLastContent.indexOf(current) >= 0) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return nodeContext.after;
+    };
+
+    /**
+     * @return {boolean}
+     */
+    RepetitiveElements.prototype.isEnableToUpdateState = function() {
+        if ((!this.isSkipFooter && this.enableSkippingFooter && this.footerViewNode)
+          || (!this.isSkipHeader && this.enableSkippingHeader && this.headerViewNode)) {
+            return true;
+        } else {
+            return false;
+        }
+    };
+
+    RepetitiveElements.prototype.updateState = function() {
+        if (!this.isSkipFooter && this.enableSkippingFooter && this.footerViewNode) {
+            this.isSkipFooter = true;
+        } else if (!this.isSkipHeader && this.enableSkippingHeader && this.headerViewNode) {
+            this.isSkipHeader = true;
+        }
+    };
+
+    RepetitiveElements.prototype.preventSkippingHeader = function() {
+        this.isSkipHeader = false;
+        this.enableSkippingHeader = false;
+    };
+    RepetitiveElements.prototype.preventSkippingFooter = function() {
+        this.isSkipFooter = false;
+        this.enableSkippingFooter = false;
+    };
+    RepetitiveElements.prototype.removeHeaderFromFragment = function() {
+        this.headerViewNodes.forEach(function(viewNode) {
+            if (viewNode.parentNode) viewNode.parentNode.removeChild(viewNode);
+        });
+        this.headerViewNodes = [];
+    };
+    RepetitiveElements.prototype.removeFooterFromFragment = function() {
+        this.footerViewNodes.forEach(function(viewNode) {
+            if (viewNode.parentNode) viewNode.parentNode.removeChild(viewNode);
+        });
+        this.footerViewNodes = [];
+    };
+
+    /**
+     * @param {Node} node
+     * @return {boolean}
+     */
+    RepetitiveElements.prototype.isHeaderViewNode = function(node) {
+        return this.headerViewNodes.some(function(viewNode) {
+            return viewNode === node;
+        });
+    };
+
+    /**
+     * @param {Node} node
+     * @return {boolean}
+     */
+    RepetitiveElements.prototype.isFooterViewNode = function(node) {
+        return this.footerViewNodes.some(function(viewNode) {
+            return viewNode === node;
+        });
+    };
+    /**
+     * @return {boolean}
+     */
+    RepetitiveElements.prototype.isHeaderRegisterd = function() {
```

Comment:
Fix it in [this commit](https://github.com/vivliostyle/vivliostyle.js/pull/323/commits/1fe20ea0018e57d094706ef6a3b2c79555dce269)